viewing_party/party.py

```diff
@@ -1,23 +1,136 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    # if any of the values in our inputs are falsy 
+    result = None
+    
+
+    if not title or not genre or not rating:
+        return None 
+    else:
+        new_movie = {}
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+        return new_movie 
+
+    
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for key in user_data["watchlist"]:
+        if key["title"] == title:
+            user_data["watchlist"].remove(key)
+            user_data["watched"].append(key)
+    return user_data

```

Comment:
It's a little dangerous to modify a collection while we are iterating over it (here,`user["watchlist"]`). In this case, it ends up being OK, since we really only expect a single copy of a movie to be in the watchlist at a time. However, if we were iterating over a list, and there was the possibility of multiple adjacent items being in the list that we want to remove, we could end up skipping over items as the list items get shifted around underneath the iteration.

If we know for certain that we only need to remove one thing, we could `break` out of the loop as soon as we've removed the one thing. Alternatively, we could find the movie record we wish to remove, and then wait to perform the `remove()` function until after completing that loop.

```suggestion
    for key in user_data["watchlist"]:
        if key["title"] == title:
            user_data["watchlist"].remove(key)
            user_data["watched"].append(key)
            break
    return user_data

```